### PR TITLE
Disable schemaType during schema edit #2197

### LINKED ISSUE #2152: Schema registry: Should not be possible to change a schema type during editing
Steps:
1. Schema 'AVRO' created
2. Open created schema
3. Select 'Edit Schema'
4. Try to change a type

Actual result:
3. It's possible to change a type of schema

Screenshot 2022-06-10 at 18 20 42

Expected result:
3. Should not be possible to change a type, so it's better to remove 'Type' from editing

### LINKED ISSUE #2197: Make a separate endpoint for updating schemas
Currently there's just one endpoint for both creating and updating a schema. Schema type param shouldn't be available when updating an existing schema.

### PR DESCRIPTION
**Pull request by shubhwip**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Disable schemaType during schema edit 
Update Schema API documentation
Closes #2197
Closes #2152

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by Haarolean** [comment]
I'm not sure about this approach. In this case, it's not obvious this param is not available for updating a schema, just for creating one. Even we got ourselves confused with our frontend part. 
If we'd stick with just one endpoint, I'd suggest making it obvious for API consumers that this param is not available for update action. Like, a hint in api declaration (kafka-ui-api.yaml) would be nice and an exception (in case there's none) upon updating.
@shubhwip what do you think?
cc: @provectus/kafka-backend

**Comment by shubhwip** [comment]
@Haarolean Nice !! I only cared for frontend initially and yes you're right it will confusing for API consumer whether they can update `schema type` field or not and they can use the same endpoint as CREATE subject for updating subject version ?
Now we can have two solutions
- As proposed by you initially a separate endpoint on `kafka-ui-api` for updating the subject and we will call same SR endpoint for updating the schema.
However we do need schema type to send along with other details otherwise SR endpoint assumes that schemaType was avro and thus it fails if schemaType was different. I tried to find a way if we can get `schemaType` from `subject name` but seems there is nothing. If there is way for us to find schemaType for given subject then we only take two params from end user i.e. `schema` and `subjectname` and find `schemaType` at our end and if we don't find we throw exception.
- We update the documentation and let API consumers know that this same endpoint will be used to update the subject version as well and you need to pass `schemaType` but make sure to not change initial `schemaType`. if a user doing it through UI then our disabled schemaType box solution will work.

**Comment by Haarolean** [comment]
@shubhwip Considering the problems you mentioned in the first approach, I'd suggest we'd go for the second one, it looks like it's a more "native" solution.

**Review by Haarolean — CHANGES_REQUESTED** [review]
backend changes required

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 1]
```diff
@@ -2594,13 +2594,15 @@ components:
...
       type: object
+      description: should be set for creating new and also updating schema subject
       properties:
         subject:
           type: string
         schema:
           type: string
         schemaType:
           $ref: '#/components/schemas/SchemaType'
+          description: should be set and provide the existing schema type when updating schema subject
```
Comment (Haarolean) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:2605]:
```suggestion
          description: upon updating a schema, the type of an existing schema can't be changed
```
made it a bit clearer

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 2]
```diff
@@ -2624,6 +2626,7 @@ components:
 
     SchemaType:
       type: string
+      description: should be set when creating/updating schema subject and provide the existing schema type when updating schema subject
```
Comment (Haarolean) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:2629]:
```suggestion
      description: upon updating a schema, the type of an existing schema can't be changed
```

**Review by Haarolean — CHANGES_REQUESTED** [review]

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 3]
```diff
@@ -2594,13 +2594,15 @@ components:
 
     NewSchemaSubject:
       type: object
+      description: should be set for create/update schema subject
```
Comment (Haarolean) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:2597]:
```suggestion
      description: should be set for creating/updating schema subject
```

**Review by Haarolean — CHANGES_REQUESTED** [review]

**Review by Haarolean — APPROVED** [review]

**Comment by Haarolean** [comment]
@shubhwip thank you :)

**Pull request merged**